data_engineering.qmd

```diff
@@ -1,121 +1,526 @@
 # Data Engineering
 
 ## Introduction
+Data is the lifeblood of AI systems. Without good data, even the most advanced machine learning algorithms will fail. In this section, we will dive into the intricacies of building high-quality datasets to fuel our AI models. Data engineering encompasses the processes of collecting, storing, processing, and managing data for training machine learning models.
 
-Explanation: This section establishes the groundwork, defining data engineering and explaining its importance and role in Embedded AI. A well-rounded introduction will help in establishing the foundation for the readers.
+Dataset creators face complex privacy and representation challenges when building high-quality training data, especially for sensitive domains like healthcare. Legally, creators may need to remove direct identifiers like names and ages. Even without legal obligations, removing such information can help build user trust. However, excessive anonymization can compromise dataset utility. Techniques like differential privacy$^{1}$, aggregation, and reducing detail provide alternatives to balance privacy and utility, but have downsides. Creators must strike a thoughtful balance based on use case.
```

Comment:
1. After the second sentence, would it be worth mentioning some big privacy laws like HIPAA or GDPR?
2. In the second to last sentence, would it be worth mentioning a few downsides? Also, one thing that made me pause was: isn't losing utility one of the downsides of maintaining privacy? When you mention downsides, does it include this loss of utility, or is it separate?

Reply:
Oh I see you mention HIPAA later in the chapter so perhaps there's no need to mention it here!